credentials/gcp.go

```
@@ -0,0 +1,171 @@
+	if err != nil {
+		return GoogleToken{}, err
+	}
+	var token GoogleToken
+	err = json.Unmarshal([]byte(userTokenJSON), &token)
+	if err != nil {
+		return GoogleToken{}, err
+	}
+
+	return token, nil
+}
+
+func (gc GoogleCredential) JSON() ([]byte, error) {
+	// This serializer method handles the override from ENV transparently
+	// to the user. We should signal that we're loading from the ENV var
+	// declared credential.
+	external, exists := os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS")
+	if exists {
+		log.Printf("Using env exported GOOGLE_APPLICATION_CREDENTIALS")
+		f, err := ioutil.ReadFile(external)
+		if err != nil {
+			log.Fatalf("Unable to read GOOGLE_APPLICATION_CREDENTIALS: %v", err)
+		}
+		return f, nil
+	}
+
+	tok, err := gc.GetCredential("default")
+	if err != nil {
+		log.Println("Unable to locate credentials. Have you run `sctl credential add`?")
+		log.Println("Another common issue is if running in a headless environment, where sctl expects" +
+			" to be invoked with GOOGLE_APPLICATION_CREDENTIALS environment variable set.")
+		log.Fatalf("Failed definetly with error %v", err)
+	}
+	return json.Marshal(tok)
+}
+
+// ValidateContext scans the environment for a declaration of authentication via ENV
+// In the case of the GCP provider, we scan for GOOGLE_APPLICAITON_CREDENTIALS and return
+// an error if the environment variable is declared.
+func (gc GoogleCredential) ValidateContext() error {
+	_, isSet := os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS")
+	if isSet {
+		return errors.New("Environment Variable GOOGLE_APPLICATION_CREDENTIALS is present, and will prevent sctl from using any user configured credentials")
+	}
+
+	return nil
+}
+
+func (gc GoogleCredential) formatCredential(token *oauth2.Token, rawClientData []byte) GoogleToken {
+	var clientJSON GoogleClientJSON
+	err := json.Unmarshal(rawClientData, &clientJSON)
+	if err != nil {
+		log.Fatalf("Error Unmarshalling client credentials: %v", err)
```

should return error instead of log.fatal